### PR TITLE
refactor: No error return for print

### PR DESCRIPTION
**Pull request by TerryHowe**
**What this PR does / why we need it**:

I don't think we should consider print errors a problem most of the time and we should manage the output of the error to print only once.

### CODE REVIEW

**Review by qweeah — CHANGES_REQUESTED** [review]

File: cmd/oras/internal/display/status/deprecated.go [hunk 1]
```diff
@@ -58,7 +58,8 @@ var printer = NewPrinter(os.Stdout)
 
 // Print objects to display concurrent-safely.
 func Print(a ...any) error {
-	return printer.Println(a...)
+	printer.Println(a...)
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply cmd/oras/internal/display/status/deprecated.go:61]: Some errors might resolve themselves like ENOSPACE could be ephemeral

File: cmd/oras/internal/display/status/print.go [hunk 2]
```diff
@@ -41,11 +43,16 @@ func NewPrinter(out io.Writer) *Printer {
...
 
 // Println prints objects concurrent-safely with newline.
-func (p *Printer) Println(a ...any) error {
+func (p *Printer) Println(a ...any) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	_, err := fmt.Fprintln(p.out, a...)
-	return err
+	if err != nil {
+		if p.errored == nil {
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply cmd/oras/internal/display/status/print.go]: Done

File: cmd/oras/internal/display/status/deprecated.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (qweeah) [reply cmd/oras/internal/display/status/deprecated.go:61]: I will resolve this comment and continue the discussion in another comment.

File: cmd/oras/internal/display/status/print.go [hunk 3]
```diff
@@ -41,11 +43,14 @@ func NewPrinter(out io.Writer) *Printer {
 }
 
 // Println prints objects concurrent-safely with newline.
-func (p *Printer) Println(a ...any) error {
+func (p *Printer) Println(a ...any) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	_, err := fmt.Fprintln(p.out, a...)
-	return err
+	if err != nil && p.errored == nil {
```
Comment (qweeah) [inline cmd/oras/internal/display/status/print.go:50]: Say we have an error `A`, which auto resolved and we have another error `B/

Should `A` and `B` be output to stderr? If so then we need to change it to
```suggestion
	if err != nil {
```

Comment (qweeah) [reply cmd/oras/internal/display/status/print.go:50]: If `&& p.errored == nil` is used then `errored` will always be wrapped `A` but not changed to `B`

**Comment by qweeah** [comment]
If `&& p.errored == nil` is not required, then `errored` can be removed from the printer like:
```golang
// Println prints objects concurrent-safely with newline.
func (p *Printer) Println(a ...any) error {
	p.lock.Lock()
	defer p.lock.Unlock()
	_, err := fmt.Fprintln(p.out, a...)
	if err != nil  {
		err = fmt.Errorf("display output error: %w", err)
		_, _ = fmt.Fprint(os.Stderr, err)
	}
        return err
}
```

File: cmd/oras/internal/display/status/print.go (hunk 3 above)
Comment (TerryHowe) [reply cmd/oras/internal/display/status/print.go:50]: It be very unusual to have A not the same as B and probably not worth distinguishing. I'm okay with last error wins, but I'd like to report it only once and it is easier to just report the first.

**Comment by TerryHowe** [comment]
Last commit only contains improved test errors

File: cmd/oras/internal/display/status/print.go (hunk 3 above)
Comment (qweeah) [reply cmd/oras/internal/display/status/print.go:50]: To me the most reasonable solution is to report all the errors.

Comment (TerryHowe) [reply cmd/oras/internal/display/status/print.go:50]: Done

Comment (qweeah) [reply cmd/oras/internal/display/status/print.go]: If we agree to report all the errors, should the error be returned directly via `printer.Println`?

Comment (qweeah) [reply cmd/oras/internal/display/status/print.go]: like
```golang
// Println prints objects concurrent-safely with newline.
func (p *Printer) Println(a ...any) error {
	p.lock.Lock()
	defer p.lock.Unlock()
	_, err := fmt.Fprintln(p.out, a...)
	if err != nil  {
		err = fmt.Errorf("display output error: %w", err)
		_, _ = fmt.Fprint(os.Stderr, err)
	}
        return err
}
```

Comment (qweeah) [reply cmd/oras/internal/display/status/print.go]: so `errored` is no longer needed.

Comment (TerryHowe) [reply cmd/oras/internal/display/status/print.go]: Sounds good to me

File: cmd/oras/internal/display/status/deprecated.go [hunk 4]
```diff
@@ -58,7 +58,8 @@ var printer = NewPrinter(os.Stdout)
...
+	return nil
```
Comment (qweeah) [inline cmd/oras/internal/display/status/deprecated.go:62]: Need documentation here to explain why and how the print errors are ignored.

File: cmd/oras/internal/display/status/print.go [hunk 5]
```diff
@@ -58,7 +62,8 @@ func (p *Printer) PrintStatus(desc ocispec.Descriptor, status string, verbose bo
 		}
 		name = desc.MediaType
 	}
-	return p.Println(status, descriptor.ShortDigest(desc), name)
+	p.Println(status, descriptor.ShortDigest(desc), name)
+	return nil
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply cmd/oras/internal/display/status/print.go:66]: `p.Println` no longer returns anything.

[the remark this answers is not in the document]
Comment (qweeah) [reply cmd/oras/internal/display/status/print.go:66]: Sorry I missed that. In my mind we should:
1) make `printer.Println` always return `nil`
2) use `return printer.Println` in `printer.PrintStatus` and `Print(a ...any)`

So the error ignoring in only implemented in `printer.Println`. In the future, if we want to change this behavior we only need to change `printer.Println`

File: cmd/oras/internal/display/status/print.go [hunk 6]
```diff
@@ -41,11 +42,14 @@ func NewPrinter(out io.Writer) *Printer {
...
+func (p *Printer) Println(a ...any) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	_, err := fmt.Fprintln(p.out, a...)
-	return err
+	if err != nil {
+		err = fmt.Errorf("display output error: %w", err)
+		_, _ = fmt.Fprint(os.Stderr, err)
+	}
 }
```
Comment (qweeah) [inline cmd/oras/internal/display/status/print.go:53]: Need to return nil here and add comment to explain why and how the print errors are ignored.

Comment (TerryHowe) [reply cmd/oras/internal/display/status/print.go:55]: I think I've handled all your concerns

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**